src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ReadMetadata.java

```diff
@@ -120,41 +187,173 @@ public ReadMetadata read( final Kryo kryo, final Input input, final Class<ReadMe
         }
     }
 
+    @DefaultSerializer(PartitionStatistics.Serializer.class)
+    public static final class PartitionStatistics {
+        private final static int MAX_TRACKED_FRAGMENT_LENGTH = 10000;
+        private final Map<String, long[]> readGroupToFragmentSizeCountMap;
+        private final long nReads;
+        private final long nBases;
+
+        public PartitionStatistics( final Iterator<GATKRead> readItr, final int nReadGroups ) {
+            readGroupToFragmentSizeCountMap = new HashMap<>(SVUtils.hashMapCapacity(nReadGroups));
+            long reads = 0L;
+            long bases = 0L;
+            while ( readItr.hasNext() ) {
+                final GATKRead read = readItr.next();
+                reads += 1L;
+                bases += read.getLength();
+                if ( read.isFirstOfPair() && !read.isSecondaryAlignment() && !read.isSupplementaryAlignment() &&
+                        !read.isUnmapped() && !read.mateIsUnmapped() &&
+                        Objects.equals(read.getContig(),read.getMateContig())) {
+                    int tLen = Math.abs(read.getFragmentLength());
+                    if ( tLen > MAX_TRACKED_FRAGMENT_LENGTH ) tLen = MAX_TRACKED_FRAGMENT_LENGTH;
```

Comment:
Could we relate the MIN_CRAZY_ZISH_SCORE above to the MAX_TRACKED_FRAGMENT_LENGHT here?